shm/player.cpp

```diff
@@ -0,0 +1,49 @@
+#include "player.hpp"
+
+Player::Player(std::unique_ptr<Ship> &ship, int money, int availableSpace)
+    : ship_(std::move(ship)), money_(money), availableSpace_(calculateAvailableSpace())
+{}
+
+std::unique_ptr<Ship> &Player::getShip()
+{
+    return ship_;
+}
+
+int Player::getMoney()
+{
+    return money_;
+}
+
+size_t Player::getAvailableSpace()
+{
+    return availableSpace_;
+}
+
+size_t Player::getSpeed() const
+{
+    return ship_->getSpeed();
+}
+
+Cargo *Player::getCargo(size_t index) const
+{
+    return ship_->getCargo()[index];
```

Comment:
What if ptr is nullptr? This function is not safe! You need to validate if you can access some cargo. Try `.at()` and  handle exception or validate size.